src/utils/connectionStringParser.ts

```diff
@@ -0,0 +1,47 @@
+import { ConnectionString, ConnectionStringKey } from '../Declarations/Contracts';
+
+import * as Constants from '../Declarations/Constants';
+
+export class ConnectionStringParser {
+  private static readonly FIELDS_SEPARATOR = ';';
+
+  private static readonly FIELD_KEY_VALUE_SEPARATOR = '=';
+
+  public static parse(connectionString?: string): ConnectionString {
+    if (!connectionString) {
+      return {};
+    }
+
+    const kvPairs = connectionString.split(ConnectionStringParser.FIELDS_SEPARATOR);
+
+    const result: ConnectionString = kvPairs.reduce((fields: ConnectionString, kv: string) => {
+      const kvParts = kv.split(ConnectionStringParser.FIELD_KEY_VALUE_SEPARATOR);
+
+      if (kvParts.length === 2) {
```

Comment:
I'm not too familiar with js syntax, is this checking that there are only 2 key value pairs, or is it checking that each entry is indeed a pair?